ldes-server-domain/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/domain/eventstream/http/services/EventStreamResponseConverterImpl.java

```diff
@@ -0,0 +1,136 @@
+package be.vlaanderen.informatievlaanderen.ldes.server.domain.eventstream.http.services;
+
+import be.vlaanderen.informatievlaanderen.ldes.server.domain.converter.PrefixAdder;
+import be.vlaanderen.informatievlaanderen.ldes.server.domain.eventstream.http.valueobjects.EventStreamResponse;
+import be.vlaanderen.informatievlaanderen.ldes.server.domain.view.service.ViewSpecificationConverter;
+import be.vlaanderen.informatievlaanderen.ldes.server.domain.viewcreation.valueobjects.AppConfig;
+import be.vlaanderen.informatievlaanderen.ldes.server.domain.viewcreation.valueobjects.ViewSpecification;
+import org.apache.jena.rdf.model.*;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static be.vlaanderen.informatievlaanderen.ldes.server.domain.constants.RdfConstants.*;
+import static org.apache.jena.rdf.model.ModelFactory.createDefaultModel;
+import static org.apache.jena.rdf.model.ResourceFactory.*;
+
+@Component
+public class EventStreamResponseConverterImpl implements EventStreamResponseConverter {
+
+	public static final String CUSTOM = "http://example.org/";
+	public static final Property MEMBER_TYPE = createProperty(CUSTOM, "memberType");
+	private final String hostname;
+	private final ViewSpecificationConverter viewSpecificationConverter;
+	private final PrefixAdder prefixAdder;
+
+	public EventStreamResponseConverterImpl(AppConfig appConfig, ViewSpecificationConverter viewSpecificationConverter,
+			PrefixAdder prefixAdder) {
+		this.viewSpecificationConverter = viewSpecificationConverter;
+		hostname = appConfig.getHostName();
+		this.prefixAdder = prefixAdder;
+	}
+
+	@Override
+	public EventStreamResponse fromModel(Model model) {
+		final String collection = getIdentifier(model, createResource(EVENT_STREAM_TYPE)).map(Resource::getLocalName)
+				.orElseThrow();
+		final String timestampPath = getResource(model, LDES_TIMESTAMP_PATH);
+		final String versionOfPath = getResource(model, LDES_VERSION_OF);
+		final List<ViewSpecification> views = getViews(model, collection);
+		final String memberType = getResource(model, MEMBER_TYPE);
+		final Model shacl = getShaclFromModel(model);
+		return new EventStreamResponse(collection, timestampPath, versionOfPath, memberType, views, shacl);
+	}
+
+	@Override
+	public Model toModel(EventStreamResponse eventStreamResponse) {
```

Comment:
This method is getting quite big. Perhaps we could extract some small private methods. Every block like:

`		eventStreamResponse.getViews().stream()
				.map(view -> createStatement(subject, createProperty(VIEW),
						viewSpecificationConverter.getIRIFromViewName(view.getName())))
				.forEach(eventStreamModel::add);`

Could be something like:

`eventStreamModel.add(getViewReferenceStatements(eventStreamResponse, subject))`

Then you can easily see high level what we are adding to the model without needing to know the details.